### PR TITLE
PLDM: Dump Offload support for non-hmc managed

### PR DESCRIPTION
**Pull request by sagisin**
Signed-off-by: Sagar Srinivas <sagar.srinivas@ibm.com>

### CODE REVIEW

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 1]
```diff
@@ -426,20 +462,63 @@ int DumpHandler::readIntoMemory(uint32_t offset, uint32_t& length,
...
 }
 
 int DumpHandler::read(uint32_t offset, uint32_t& length, Response& response,
                       oem_platform::Handler* /*oemPlatformHandler*/)
 {
-    if (dumpType != PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    auto path = findDumpObjPath(fileHandle);
+    static constexpr auto dumpFilepathInterface =
+        "xyz.openbmc_project.Common.FilePath";
+    if (dumpType == PLDM_FILE_TYPE_DUMP)
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io_type_dump.cpp:500]: This has changed the old logic. Add an || with PLDM_FILE_TYPE_RESOURCE_DUMP too.

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 2]
```diff
@@ -426,20 +462,63 @@ int DumpHandler::readIntoMemory(uint32_t offset, uint32_t& length,
...
+                pldm::utils::DBusHandler().getDbusProperty<std::string>(
+                    path.c_str(), "Path", dumpFilepathInterface);
+            auto rc = readFile(filePath.c_str(), offset, length, response);
+            return rc;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Failed to fetch the filepath of the dump entry"
+                      << std::hex << fileHandle << ", error = " << e.what()
+                      << "\n";
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io_type_dump.cpp:518]: Please add pldm::utils::reportError()

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 3]
```diff
@@ -426,20 +462,63 @@ int DumpHandler::readIntoMemory(uint32_t offset, uint32_t& length,
...
+                    path.c_str(), "Path", dumpFilepathInterface);
+            auto rc = transferFileData(fs::path(filePath), false, offset,
+                                       length, address);
+            return rc;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Failed to fetch the filepath of the dump entry"
+                      << std::hex << fileHandle << ", error = " << e.what()
+                      << "\n";
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io_type_dump.cpp:487]: Please add pldm::utils::reportError()

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 4]
```diff
@@ -53,6 +59,31 @@ std::string DumpHandler::findDumpObjPath(uint32_t fileHandle)
...
+    {
+        curResDumpEntryPath =
+            (std::string)bmcDumpObjPath + "/" + std::to_string(fileHandle);
+        return curResDumpEntryPath;
+    }
+    else if (dumpType == PLDM_FILE_TYPE_SBE_DUMP)
+    {
+        curResDumpEntryPath =
+            (std::string)sbeDumpObjPath + "/" + std::to_string(fileHandle);
+        return curResDumpEntryPath;
```
[the remark this answers is not in the document]
Comment (sagisin) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp:72]: The flow after this code block corresponds to the normal flow for resource/system dump. should we go through the entire flow if we are getting curResDumpEntryPath earlier?

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 5]
```diff
@@ -426,20 +462,63 @@ int DumpHandler::readIntoMemory(uint32_t offset, uint32_t& length,
                                 uint64_t address,
                                 oem_platform::Handler* /*oemPlatformHandler*/)
 {
-    if (dumpType != PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    auto path = findDumpObjPath(fileHandle);
+    static constexpr auto dumpFilepathInterface =
+        "xyz.openbmc_project.Common.FilePath";
+    if (dumpType == PLDM_FILE_TYPE_DUMP)
```
[the remark this answers is not in the document]
Comment (sagisin) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp:468]: done

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp (hunk 2 above)
Comment (sagisin) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp:518]: done

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp (hunk 3 above)
Comment (sagisin) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp:487]: done

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (manojkiraneda) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp:72]: what i was saying is return curResDumpEntryPath at line number 86  , as everything is in else if we don't enter into other conditions right ? i don't see how moving that could cause issues.

[the remark this answers is not in the document]
Comment (sagisin) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp]: done

**Review by jaypadath — APPROVED** [review]

**Review by manojkiraneda — APPROVED** [review]

**Pull request merged**